File: app/src/main/java/com/turlir/abakcalc/DirectPrinter.java
package com.turlir.abakcalc;

import com.turlir.converter.Printer;

import java.math.BigDecimal;
import java.text.DecimalFormat;

public class DirectPrinter implements Printer {

    private final StringBuilder mBuilder;
    private final DecimalFormat mFormat;

    public DirectPrinter(DecimalFormat format) {
        mFormat = format;
        mBuilder = new StringBuilder();
    }

    @Override
    public void append(String part) {
        mBuilder.append(part);
    }

    @Override
    public void append(BigDecimal part) {
        String str = mFormat.format(part);
        append(str);
    }

    @Override
    public void appendSeparator() {
        append(String.valueOf(mFormat.getDecimalFormatSymbols().getDecimalSeparator()));
    }

    @Override
    public int length() {
        return mBuilder.length();
    }

    @Override
    public void reset() {
        mBuilder.delete(0, length());
    }

    @Override
    public String toString() {
        return mBuilder.toString();
    }
}

File: app/src/main/kotlin/com/turlir/abakcalc/DirectPrinter.kt
package com.turlir.abakcalc

import com.turlir.calculator.converter.Printer

import java.math.BigDecimal
import java.text.DecimalFormat

class DirectPrinter(private val mFormat: DecimalFormat) : Printer {

    private val mBuilder = StringBuilder()

    override fun append(part: String) {
        mBuilder.append(part)
    }

    override fun append(part: BigDecimal) {
        val str = mFormat.format(part)
        append(str)
    }

    override fun appendSeparator() {
        append(mFormat.decimalFormatSymbols.decimalSeparator.toString())
    }

    override fun length() = mBuilder.length

    override fun reset() {
        mBuilder.delete(0, length())
    }

    override fun toString() = mBuilder.toString()
}
